addons/account/models/account_move.py

```diff
@@ -2997,10 +2997,11 @@ def default_get(self, default_fields):
     def name_get(self):
         result = []
         for line in self:
-            if line.ref:
-                result.append((line.id, (line.move_id.name or '') + '(' + line.ref + ')'))
-            else:
-                result.append((line.id, line.move_id.name))
+            name = line.move_id.name or ''
+            ref = line.ref or line.name or line.product_id.display_name
+            if ref:
+                name += " (%s)" % ref
+            result.append((line.id, name))
```

Comment:
I'd rather have something like 
```python
            name = line.move_id.name or ''
            if line.ref:
                name += " (%s)" % line.ref
            name += ' '.join(line.name or line.product_id.display_name or '')
            result.append((line.id, name))
```
because when a move has a ref, it will be copied on all its move lines, hence still no good way to determine what line to choose in a m2o

Reply:
done

Reply:
```python
name += ' '.join(line.name or line.product_id.display_name or '')
```
is plain wrong
```python
name += (line.name or line.product_id.display_name) and ' ' + (line.name or line.product_id.display_name) or ''
```

Reply:
done?